projects/pre-sentence-reports-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/document/DocumentService.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.justice.digital.hmpps.integrations.delius.document
+
+import org.springframework.http.HttpEntity
+import org.springframework.http.MediaType
+import org.springframework.http.client.MultipartBodyBuilder
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.util.MultiValueMap
+import uk.gov.justice.digital.hmpps.audit.service.AuditableService
+import uk.gov.justice.digital.hmpps.audit.service.AuditedInteractionService
+import uk.gov.justice.digital.hmpps.config.ServiceContext
+import uk.gov.justice.digital.hmpps.exception.ConflictException
+import uk.gov.justice.digital.hmpps.exception.NotFoundException
+import uk.gov.justice.digital.hmpps.integrations.alfresco.AlfrescoClient
+import uk.gov.justice.digital.hmpps.integrations.delius.audit.BusinessInteractionCode
+import uk.gov.justice.digital.hmpps.integrations.delius.courtreport.CourtReportRepository
+import uk.gov.justice.digital.hmpps.message.AdditionalInformation
+import uk.gov.justice.digital.hmpps.message.SimpleHmppsEvent
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+
+@Service
+class DocumentService(
+    auditedInteractionService: AuditedInteractionService,
+    private val documentRepository: DocumentRepository,
+    private val courtReportRepository: CourtReportRepository,
+    private val alfrescoClient: AlfrescoClient
+) : AuditableService(auditedInteractionService) {
+    fun AdditionalInformation.reportId() = this["reportId"] as String
+    @Transactional
+    fun updateCourtReportDocument(hmppsEvent: SimpleHmppsEvent, file: ByteArray) =
+        audit(BusinessInteractionCode.UPLOAD_DOCUMENT) {
+            val reportId = hmppsEvent.additionalInformation.reportId()
+            val document = documentRepository.findByExternalReference(reportId)
+                ?: throw NotFoundException("Document", "externalReference", reportId)
+            it["documentId"] = document.id
+            it["alfrescoDocumentId"] = document.alfrescoId
```

Comment:
Currently we also audit the details of what the document is attached to - would make sense to do the same here too:
```suggestion
            it["documentId"] = document.id
            it["alfrescoDocumentId"] = document.alfrescoId
            it["entityId"] = document.courtReportId
            it["tableName"] = "COURT_REPORT"
```

Reply:
Isn't this the relevant call in Delius API?  Maybe worth confirming we haven't missed anything else in there if so.
 https://github.com/ministryofjustice/hmpps-delius-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/service/court/report/CourtReportService.kt#L72-L74